src/v/storage/disk_log_impl.cc

```diff
@@ -1969,6 +1969,198 @@ ss::future<log::offset_range_size_result_t> disk_log_impl::offset_range_size(
     };
 }
 
+ss::future<log::offset_range_size_result_t> disk_log_impl::offset_range_size(
```

Comment:
Can we please break stuff like this out into separate PRs along with unit tests? IIUC this type of change is a separate, pre-requisite for the upload mechanism work?

Reply:
done